apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -1007,4 +1008,32 @@ export class ApplicationController {
 
     return { url }
   }
+
+  @Scopes(ApplicationScope.read)
+  @Get('applications/:id/attachments/:key')
+  @ApiParam({
+    name: 'id',
+    type: String,
+    required: true,
+    description: 'The id of the application to update the attachments for.',
+    allowEmptyValue: false,
+  })
+  @ApiOkResponse({ type: FileContentAsBase64ResponseDto })
+  async getFileContentAsBase64(
+    @Param('id', new ParseUUIDPipe()) id: string,
+    @Param('key') key: string,
+    @CurrentUser() user: User,
+  ): Promise<FileContentAsBase64ResponseDto> {
+    const existingApplication = await this.applicationAccessService.findOneByIdAndNationalId(
+      id,
+      user.nationalId,
+    )
```

Comment:
Unfortunately, this opens up a big hole in our application system. Anyone who has some access to an application would be able to access and download any (sensitive) files associated with it. This is why I asked that you would sync up before going forward with this change. :)

Basically, we need some way to access control this.

My first idea is to use the existing read/write access control system and allow the user to access files which are referred to in "answers" which they have access to (eg `GET /applications/:id/attachment?field=photos[0]`). This way you can download attachments you uploaded and any other you "have access to". But we need to make sure the user can't change some answer to download files they shouldn't have access to.

The attachment keys include a 122bit random uuidv4, making them pretty much impossible to brute force (with our API rate limits). However, the `application.attachments` field contains a list of all of the attachment keys. No client should need it or should be using it currently, but currently it is queried and available **everywhere**. So we'd need to remove it from all API responses (both Application System API and GraphQL)... 

That's "the project" (and what makes this take a week or more).

Happy to discuss alternative ways.